src/nvim/screen.c

```diff
@@ -7505,6 +7505,10 @@ void screen_resize(int width, int height)
   ++busy;
 
   Rows = height;
+  int max_p_ch = Rows - min_rows() + 1;
```

Comment:
you want this check below `check_shellsize()`, that should fix the text failure.

Reply:
@bfredl thanks. With that change, it now passes the test locally.